requests/script.js

```diff
@@ -421,6 +459,55 @@ async function renderOooRequestCards(queries = {}) {
   nextLink = oooRequestResponse?.next;
 }
 
+// Function to get extension requests
+async function getExtensionRequests(query = {}) {
+  if (requestController) {
+    requestController.abort();
+    requestController = new AbortController();
+  }
+  let finalUrl =
+    API_BASE_URL +
+    (nextLink || '/extension-requests' + getExtensionQueryParamsString(query));
+  let windowUrl = `${window.location.origin}${window.location.pathname}`;
+
+  window.history.pushState({ path: windowUrl }, '', windowUrl);
+  try {
+    const res = await fetch(finalUrl, {
+      credentials: 'include',
+      signal: extensionRequestController.signal,
+    });
+    const data = await res.json();
+    if (res.ok) {
+      return data;
```

Comment:
the fetch logic in `getOooRequests` and `getExtensionRequests` can be abstracted into a reusable function.